packages/@react-spectrum/textfield/src/TextArea.tsx

```diff
@@ -25,18 +25,32 @@ function TextArea(props: SpectrumTextFieldProps, ref: RefObject<TextFieldRef>) {
     isReadOnly = false,
     isRequired = false,
     onChange,
+    value,
+    defaultValue,
     ...otherProps
   } = props;
 
   let inputRef = useRef<HTMLInputElement & HTMLTextAreaElement>();
 
-  let onHeightChange = () => {
+  let onHeightChange = useCallback(() => {
     if (isQuiet) {
       let input = inputRef.current;
       input.style.height = 'auto';
       input.style.height = `${input.scrollHeight}px`;
     }
-  };
+  }, [isQuiet, inputRef]);
+
+  useEffect(() => {
+    if (inputRef.current) {
+      // if no value or defaultValue is passed
+      // no need to call onHeightChange
+      if (!value && !defaultValue) {
```

Comment:
Perhaps this if statement should be something along the lines of 
```suggestion
      if (!inputRef.current.value) {
```
instead? This would make `onHeightChange` properly fire for cases where a non-quiet TextArea with text within it is changed to quiet (in the current implementation it is isn't fired because `value` and `defaultValue` are undefined even if a user has typed text into the field).

Maybe this check could be removed entirely and the failing test is rewritten to something like:
```
    expect(input.style.height).toBe(`${mockScrollHeight}px`);
    Object.defineProperty(HTMLElement.prototype, 'scrollHeight', {configurable: true, value: 1000});
    fireEvent.change(input, {target: {value: '15', style: {}}});
    expect(input.style.height).toBe(`1000px`);
```
since we are really just testing that onHeightChange is fired when the input field receives a change event. What do you think @snowystinger ?

Reply:
yeah, I don't think this is enough reason to add a full stately package... so `inputRef.current.value` is probably good to start. we can discuss further if need arises.
thanks for catching that case